worlds/Files.py

```diff
@@ -355,6 +356,8 @@ def write_token(self, token_type: APTokenTypes, offset: int, data: Union[bytes,
         """
         Stores a token to be used by patching.
         """
+        if not getattr(self, "tokens", None):
```

Comment:
Also I completely forgot about hasattr.
Your `getattr(...)` would return `False` for `tokens = []` while `hasattr(...)` would return `True`. Not sure which is better. `getattr` could hide mistakes while `hasattr` is more likely in accidental sharing of class vars again. I feel like `hasattr` would be expected here though.

Reply:
I would agree I think. I mostly used `getattr` because I forgot `hasattr` existed.